script.js

```diff
@@ -1,4 +1,116 @@
+var gameMode = "Lets start the game";
+var numberOfPlayers = 0;
+var numberOfDice = 0;
+var playerAnswer = [];
+var sumOfPlayerScore = [];
+var playerArray = [];
+var curPlayer = 1;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (gameMode == "Lets start the game") {
+    gameMode = "Input number of players";
+    console.log(gameMode);
+    return "Welcome to Beat That! Please input number of players.";
+  }
+  if (gameMode == "Input number of players") {
+    numberOfPlayers = input;
+    if (Number.isNaN(Number(input))) {
+      return `Sorry please enter a number`;
+    } else {
+      getPlayerArray();
+      console.log(numberOfPlayers);
+      gameMode = "Input number of dice";
+      console.log(gameMode);
+      return "Please input number of dice want to play.";
+    }
+  }
+  if (gameMode == "Input number of dice") {
+    numberOfDice = input;
+    if (Number.isNaN(Number(input))) {
+      return `Sorry please enter a number`;
+    } else {
+      gameMode = "Roll the dice";
+      console.log(gameMode);
+      console.log(numberOfDice);
+      return "Press continue to roll the dice";
+    }
+  }
+  if (gameMode == "Roll the dice") {
+    var diceResult = displayRolledResult();
+    gameMode = "Key in answer";
+    console.log(gameMode);
+    return diceResult + "<br>" + "Please input your answer.";
+  }
+  if (gameMode == "Key in answer") {
+    sumOfPlayerScore[Number(curPlayer - 1)] += Number(input);
+    gameMode = "Roll the dice";
+    curPlayer += 1;
+    if (curPlayer > numberOfPlayers) {
+      gameMode = "leaderboard";
+      return `All players have input the answer. Press continue to see the result❗`;
+    }
```

Comment:
Might be possible to abstract out the logic in each if block into it's own function, e.g. lines 45-50 can be in a function `executeKeyInAnswer`